responses/40-datasci-conventions.md

```diff
@@ -0,0 +1,61 @@
+
+What you’ve learned so far are about the mechanics of using the package, but there are a few conventions that USGS Data Science practioners use to maintain some standards across projects, which make it easier to jump into a new project, provide peer review, or learn a new technique from someone else’s pipeline since you are already familiar with the structure.
+
+---
+
+As you learned in the first pipelines course, we like to separate pipelines into distinct phases based on what is happening to the data (we usually use `fetch`, `process`, `visualize`, etc). So far in this course, we have been using a single list of targets in the `_targets.R` makefile. This works for short pipelines, but when you have bigger, more complex pipelines, that file and target list could get HUGE and difficult to read. 
+
+For this reason, we like to have multiple makefiles that are sourced by the "canonical" makefile (the one named `_targets.R`). In each one of these makefiles, targets are saved in an R list object which is named and numbered based on the phase, e.g. `p1_fetch`. Then, the main `_targets.R` makefile combines those into a single list using `c()`.
```

Comment:
```suggestion
For this reason, we like to have multiple makefiles that each describe and are named after a single phase, e.g., `1_fetch.R` or `4_visualize.R`. Within each phase makefile, targets are saved in an R list object which is numbered based on the phase, e.g. `p1_targets` or `p4_targets`. Then, the main `_targets.R` makefile sources each of these phase makefiles and combines the target lists into a single list using `c()`, e.g., `c(p1_targets, p2_targets, p3_targets, p4_targets`.
```

In the past, I've just used `p1`, `p2`, etc. as the names of the target lists to concatenate within `_targets.R` - I like that my approach reduces redundancy such that if you renamed `1_fetch.R` you wouldn't also need to rename the list of targets, and it's totally copy-pasteable from one pipeline to another (except for adjusting the number of `pX` objects that get concatenated). What do you think? Alternatively, we could use `p1_targets`, `p2_targets` to communicate the nature/contents of those R lists - did Jake suggest something like this a while ago? What do you think about the `p1` or `p1_targets` approaches as alternatives to `p1_fetch`?

Reply:
Hmmm I was following this discussion https://code.usgs.gov/wma/wp/forecasted-met-drivers/-/merge_requests/1#note_128166 and I think I missed Jake's final comment! I think your changes here reflects the consensus from that conversation.

Reply:
Glad you found it, and thanks for adding the link. Cool, let's go with `p1_targets_list` to stick with what we agreed with Jake.